Sources/CoreMetrics/SystemMetrics.swift

```diff
@@ -0,0 +1,204 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Metrics API open source project
+//
+// Copyright (c) 2018-2020 Apple Inc. and the Swift Metrics API project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Metrics API project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+#if os(Windows)
+public typealias ProcessId = DWORD
```

Comment:
why don't we just make them `Int`? That's the recommendation in the [Swift language guide](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html#ID324).

> Numeric Type Conversion
>
> Use the Int type for all general-purpose integer constants and variables in your code, even if they’re known to be nonnegative. Using the default integer type in everyday situations means that integer constants and variables are immediately interoperable in your code and will match the inferred type for integer literal values.
>
> Use other integer types only when they’re specifically needed for the task at hand, because of explicitly sized data from an external source, or for performance, memory usage, or other necessary optimization. Using explicitly sized types in these situations helps to catch any accidental value overflows and implicitly documents the nature of the data being used.
>

Reply:
should we be worried that `Int` is too small on 32 bit platforms, then we should go with `Int64` IMHO

Reply:
> why don't we just make them `Int`? That's the recommendation in the [Swift language guide](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html#ID324).
> 
> 

The recommendation didn’t work well here.  A DWORD is the system type for this and is not an Int.  Using Int will cause type conversations which need to be made explicit.  I’m strongly against using Int for the process type.


> 
> > Numeric Type Conversion
> 
> >
> 
> > Use the Int type for all general-purpose integer constants and variables in your code, even if they’re known to be nonnegative. Using the default integer type in everyday situations means that integer constants and variables are immediately interoperable in your code and will match the inferred type for integer literal values.
> 
> >
> 
> > Use other integer types only when they’re specifically needed for the task at hand, because of explicitly sized data from an external source, or for performance, memory usage, or other necessary optimization. Using explicitly sized types in these situations helps to catch any accidental value overflows and implicitly documents the nature of the data being used.
> 
> >

Reply:
@compnerd / @MrLotU they can't be different on different platforms because it's a `public protocol`. Otherwise you'd get implementations that only confirm on say Linux and not on Windows.

Reply:
They are different across the platforms.  I suppose that crafting a wrapper that is constructed by the platforms type should alleviate your concern.

Reply:
wrapper works but I was thinking of `Int` as the "wrapper". `Int(myDWORD)` works, just like `Int(my_pid_t)` does.